src/test_l3_dsHdmiIn.c

```diff
@@ -0,0 +1,1429 @@
+  { "dsAVICONTENT_TYPE_GAME",        (int32_t)dsAVICONTENT_TYPE_GAME     },
+  { "dsAVICONTENT_TYPE_INVALID",     (int32_t)dsAVICONTENT_TYPE_INVALID  },
+  { "dsAVICONTENT_TYPE_MAX",         (int32_t)dsAVICONTENT_TYPE_MAX      },
+  {  NULL, -1 }
+};
+
+/* dsVideoPlaneType_t */
+const static ut_control_keyMapTable_t dsVideoPlaneType_mapTable [] = {
+  { "dsVideoPlane_PRIMARY",     (int32_t)dsVideoPlane_PRIMARY   },
+  { "dsVideoPlane_SECONDARY",   (int32_t)dsVideoPlane_SECONDARY },
+  { "dsVideoPlane_MAX",         (int32_t)dsVideoPlane_MAX       },
+  {  NULL, -1 }
+};
+
+/* dsVideoZoom_t */
+const static ut_control_keyMapTable_t dsVideoZoom_mapTable [] = {
+  { "dsVIDEO_ZOOM_UNKNOWN",          (int32_t)dsVIDEO_ZOOM_UNKNOWN           },
+  { "dsVIDEO_ZOOM_NONE",             (int32_t)dsVIDEO_ZOOM_NONE              },
+  { "dsVIDEO_ZOOM_FULL",             (int32_t)dsVIDEO_ZOOM_FULL              },
+  { "dsVIDEO_ZOOM_LB_16_9",          (int32_t)dsVIDEO_ZOOM_LB_16_9           },
+  { "dsVIDEO_ZOOM_LB_14_9",          (int32_t)dsVIDEO_ZOOM_LB_14_9           },
+  { "dsVIDEO_ZOOM_CCO",              (int32_t)dsVIDEO_ZOOM_CCO               },
+  { "dsVIDEO_ZOOM_PAN_SCAN",         (int32_t)dsVIDEO_ZOOM_PAN_SCAN          },
+  { "dsVIDEO_ZOOM_LB_2_21_1_ON_4_3", (int32_t)dsVIDEO_ZOOM_LB_2_21_1_ON_4_3  },
+  { "dsVIDEO_ZOOM_LB_2_21_1_ON_16_9",(int32_t)dsVIDEO_ZOOM_LB_2_21_1_ON_16_9 },
+  { "dsVIDEO_ZOOM_PLATFORM",         (int32_t)dsVIDEO_ZOOM_PLATFORM          },
+  { "dsVIDEO_ZOOM_16_9_ZOOM",        (int32_t)dsVIDEO_ZOOM_16_9_ZOOM         },
+  { "dsVIDEO_ZOOM_PILLARBOX_4_3",    (int32_t)dsVIDEO_ZOOM_PILLARBOX_4_3     },
+  { "dsVIDEO_ZOOM_WIDE_4_3",         (int32_t)dsVIDEO_ZOOM_WIDE_4_3          },
+  { "dsVIDEO_ZOOM_MAX",              (int32_t)dsVIDEO_ZOOM_MAX               },
+  {  NULL, -1 }
+};
+
+/* tv_hdmi_edid_version_t */
+const static ut_control_keyMapTable_t tv_hdmi_edid_version_mapTable [] = {
+  { "HDMI_EDID_VER_14",     (int32_t)HDMI_EDID_VER_14  },
+  { "HDMI_EDID_VER_20",     (int32_t)HDMI_EDID_VER_20  },
+  { "HDMI_EDID_VER_MAX",    (int32_t)HDMI_EDID_VER_MAX },
+  {  NULL, -1 }
+};
+
+/* bool */
+const static ut_control_keyMapTable_t bool_mapTable [] = {
+  { "false", (int32_t)false },
+  { "true",  (int32_t)true  },
+  {  NULL, -1 }
+};
+
+/**
+ * @brief This functions gets the Enum mapping string.
+ *
+ * This functions gets the Enum mapping string.
+ * &Todo: replace with ut control function
+ */
+static char* ut_control_GetMapString(const ut_control_keyMapTable_t *conversionMap, int32_t key)
+{
+    int32_t count = 0;
+    while(conversionMap[count].string != NULL){
+        if(conversionMap[count].key == key) {
+            return conversionMap[count].string;
+        }
+        count++;
+    }
+    return "";
+}
+
+/**
+ * @brief This function clears the stdin buffer.
+ *
+ * This function clears the stdin buffer.
+ */
+void readAndDiscardRestOfLine(FILE* in)
+{
```

Comment:
void list_ports() - seeing the function in the name. Add in the function naming way like listPorts()

Reply:
done